src/final_aggregate_analysis.ts

```diff
@@ -0,0 +1,625 @@
+import * as fs from 'fs';
+import * as util from 'util';
+import {
+  ALL_SQUARES,
+  Piece,
+  PrettyMove,
+  Square,
+  UASymbol,
+} from '../cjsmin/src/chess';
+// const processFiles = require('./streaming_partial_decompresser.js');
+
+const readFile = util.promisify(fs.readFile);
+
+/**
+ *
+ * @param results.json
+ * @returns final analysis results of all files created and deleted in streaming_partial_decompresser
+ */
+async function aggregateResults(filePath: string) {
+  const data = JSON.parse(await readFile(filePath, 'utf-8'));
```

Comment:
yes, see line 613: `aggregateResults('src/results.json');`

The idea is that streaming_partial_decompresser.js pushes all end data into results.json, and then aggregateResults combines all that data into one takeaway

Reply:
true re: fs.readFileSync, just depends if we want this to be synchronous or not. Given your idea to prevent overwriting by writing to multiple results.json files, might be better to keep reading from those results files async, which is what is currently implemented by promisify